docs/community/maintainers-guide/releasing.rst

```diff
@@ -0,0 +1,224 @@
+..
+    This file is part of Invenio.
+    Copyright (C) 2020 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+Releasing
+=========
+
+In general, you have two types of releases:
+
+- **Major/Minor-level release** - Often these releases add significant new
+  features and needs to go through extensive testing.
+- **Patch-level release** - Often these releases fix security issues or
+  important bugs.
+
+For major/minor-level releases, you'll often be working only with the
+``master``-branch and only make a single PyPI release. For patch-level
+releases you'll often be working with multiple maintenance branches and make
+several PyPI releases for all currently maintained versions of Invenio.
+
+Depending on the package that you are releasing, you will have to follow
+a different procedure. You might have to release:
+
+- one of the Python or JavaScript/React modules (happening rather often)
+- the Invenio framework, the ``invenio`` package (happening less often)
+
+Before you start
+----------------
+Prior to preparing a new Invenio release, please ensure you have the latest
+changes in your local environment for the maintenance branches and ``master``:
+
+1. Check that all relevant PRs have been merged. For the ``invenio`` package,
+   check that all relevant Invenio modules have been released.
+2. Fetch latest changes:
+    - ``git fetch inveniosoftware``
+3. Checkout maintenance or master branch:
+    - ``git checkout maint-<major>.<minor>``
+4. Merge or reset to latest changes:
+    - ``git merge --ff-only inveniosoftware/maint-<major>.<minor>``
+    - or
+    - ``git reset --hard inveniosoftware/maint-<major>.<minor>``
```

Comment:
Minor: personally I would add a `git reset --hard` so easily copy-pastable, I would rather say between parenthesis "if you cannot fast-forward merge make sure you save your local changes in a different branch if you need to, and reset to `inveniosoftware/maint-<major>.<minor>`" or shorter message.

Reply:
Im on team Nico on this one. One possible problem I see with having a simple `git reset --hard` copy-pastable is that if not properly configured it can actually reset all branches, and it is sort of a destructive action. In addition the rest of the docs follo the <> in terms of "fill with your concrete case"

Reply:
That is precisely what I originally meant if it wasn't because of a very tricky typo (see that if you read it now it makes more sense the whole sentence since I say "just say it in words and let the person to manage as they know"):
```diff
- Minor: personally I would add a git reset --hard so easily copy-pastable, I would rather say between parenthesis "if you cannot fast-forward merge make sure you save your local changes in a different branch if you need to, and reset to inveniosoftware/maint-<major>.<minor>" or shorter message.
+ Minor: personally I wouldn't add a git reset --hard so easily copy-pastable, I would rather say between parenthesis "if you cannot fast-forward merge make sure you save your local changes in a different branch if you need to, and reset to inveniosoftware/maint-<major>.<minor>" or shorter message.
```

Reply:
Fixed